src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -493,6 +493,10 @@ HeapDumpDCmd::HeapDumpDCmd(outputStream* output, bool heap) :
 void HeapDumpDCmd::execute(DCmdSource source, TRAPS) {
   jlong level = -1; // -1 means no compression.
   jlong parallel = HeapDumper::default_num_of_dump_threads();
+  // in case no specific filename was passed to heap_dump command,
+  // attempt to use the HeapDumpPath if it was set
+  const char* filename = HeapDumpPath;
```

Comment:
Why is this here? It is confusing because it makes it look like the HeapDumpPath will be used in place of specifying a filename with the jcmd, but in fact HeapDumpPath can be null, and logic below relies on that fact. Why not just check if HeapDumpPath is null below. You don't really need `filename`.  See suggested changes below.